### PR TITLE
Rename reflect 'hash' method to 'reflect_hash'

### LINKED ISSUE #943: Reflect::hash collides with Hash::hash
With the merging of @cart's #926, the `hash` method of `bevy::prelude::Reflect` now collides with the `hash` method from `std::hash::Hash`.  You can workaround by disambiguating, so this isn't a showstopper.  I'm reporting it in the hopes that there's an easy tweak to avoid having to do the workaround. If not, no big deal - and I'm fine closing this.

With the following imports:

```rust
use bevy::prelude::*;
use std::hash::Hash;
```

When you attempt to call the `hash` method on a primitive value like an integer (which is something you might want to do within an implementation of the Hash trait for a struct, such as on line 59 here -- `PlayerID` is a newtype over `usize`):

https://github.com/CleanCut/punchball/blob/9f8ddb2de2b8a91629307ee81b3f64aaa1eed1de/src/player/collision.rs#L52-L62

You get this error:

```
error[E0034]: multiple applicable items in scope
  --> src/player/collision.rs:58:23
   |
58 |             player_id.hash(state);
   |                       ^^^^ multiple `hash` found
   |
   = note: candidate #1 is defined in an impl of the trait `Hash` for the type `usize`
   = note: candidate #2 is defined in an impl of the trait `bevy::prelude::Reflect` for the type `usize`
help: disambiguate the associated function for candidate #1
   |
58 |             Hash::hash(&player_id, state);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: disambiguate the associated function for candidate #2
   |
58 |             bevy::prelude::Reflect::hash(&player_id, state);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0034`.
```

The workaround is to change `player_id.hash(state);` to `Hash::hash(&player_id, state);` to disambiguate.

https://github.com/CleanCut/punchball/blob/a6abd16d239fb7c6ea2656990c6da16056d4ffce/src/player/collision.rs#L52-L62

### PR DESCRIPTION
**Pull request by CleanCut**
@cart is this what you had in mind to resolve #943? (It works for me.)

- Renamed `hash` method to `reflect_hash`
- Renamed `partial_eq` method to `reflect_partial_eq`

### CODE REVIEW

File: crates/bevy_reflect/src/reflect.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ pub trait Reflect: Any + Send + Sync {
     fn reflect_mut(&mut self) -> ReflectMut;
     fn clone_value(&self) -> Box<dyn Reflect>;
     /// Returns a hash of the value (which includes the type) if hashing is supported. Otherwise `None` will be returned.
-    fn hash(&self) -> Option<u64>;
+    fn reflect_hash(&self) -> Option<u64>;
     /// Returns a "partial equal" comparison result if comparison is supported. Otherwise `None` will be returned.
     fn partial_eq(&self, _value: &dyn Reflect) -> Option<bool>;
```
Comment (cart) [inline crates/bevy_reflect/src/reflect.rs:35]: I think it probably makes sense to also rename `partial_eq` to `reflect_partial_eq`, just for consistency.

**Comment by cart** [comment]
Yeah this definitely resolve #943 in my mind / is my ideal solution to the problem

File: crates/bevy_reflect/src/reflect.rs (hunk 1 above)
Comment (CleanCut) [reply crates/bevy_reflect/src/reflect.rs]: ✅ Done and pushed.

**Comment by cart** [comment]
Hexasphere pushed a breaking minor version change (which broke ci). We've updated master, so a rebase should fix the problem.

**Review by cart — APPROVED** [review]

**Comment by CleanCut** [comment]
I rebased and force-pushed...but CI is just sitting there and not starting. 🤷‍♂️

**Comment by cart** [comment]
looks like it finally worked itself out!

**Pull request merged**